### ISSUE TITLE
How to disable devtools ?

### ISSUE THREAD

**Issue by Mikaleb** [vuejs/devtools-next#406]
Is there an option to disable this shiny new devtool ?

**Comment by Mikaleb** [comment]
Perfect thanks !

**Comment by alexzhang1030** [comment]
Can you unregister `vite-plugin-vue-devtools` in plugin options of `vite.config.{j,t}s`?

**Comment by gangsthub** [comment]
> Is there an option to disable this shiny new devtool ?

What is the reason to disable it and what are the conditions to either show it or not? I think you could provide more context about what is it what you expect, @Mikaleb...

**Comment by alexzhang1030** [comment]
Currently, DevTools will be forcibly enabled during development (vite dev) and will never be bundled in production (vite build). So, don't worry about that.

**Comment by alexzhang1030** [comment]
> > Currently, DevTools will be forcibly enabled during development (vite dev) and will never be bundled in production (vite build). So, don't worry about that.
> 
> maybe he want to disable the tool, when the envirment is develop.
> 
> just use shotcut to show tool's UI, but don't show the icon always.

<img width="1261" alt="image" src="https://github.com/vuejs/devtools-next/assets/49969959/82677dd6-e499-4c43-9e00-0ab90b502954">

You can disable this option to do this.

**Comment by eachann1024** [comment]
> > Is there an option to disable this shiny new devtool ?
> 
> What is the reason to disable it and what are the conditions to either show it or not? I think you could provide more context about what is it what you expect, @Mikaleb...

I think the corresponding habit settings can be saved in the project's `.local`, for example:
1. Scenario 1: Enter pinia, open the store / time by default.
2. Project members prefer the chrome extension version of vue dev. You can set whether to enable it in `.local`.

**Comment by gazedash** [comment]
> You can disable this option to do this. 

How to open it again? Lmao... It disappeared and I don't get it how to open it back again.

First, I try to search HOW TO DISABLE IT, spend 10 minutes on it, find old solution, doesn't work
Find this, fine, toggle this thingy, it disappears completely

Are you serious?

I understand that devtools is a good and helpful thing, but UX is severely lacking on this one...

**Comment by alexzhang1030** [comment]
> > You can disable this option to do this.
> 
> How to open it again? Lmao... It disappeared and I don't get it how to open it back again.
> 
> First, I try to search HOW TO DISABLE IT, spend 10 minutes on it, find old solution, doesn't work Find this, fine, toggle this thingy, it disappears completely
> 
> Are you serious?
> 
> I understand that devtools is a good and helpful thing, but UX is severely lacking on this one...

Yeah, our shortcut descriptions are very lacking 😅. And sorry our contributors are currently very busy and not too active in solving this issue 🥲.

Anyway, this is a brief:

### How to disable the devtools anchor?

You can disable the option called `Always show the floating panel` on the `Settings` page.

### How to open it again?

By shortcut 

- `cmd + shift + D` in `macOS`
- `alt + shift + D` in `Windows`

 align with `Nuxt devtools`.

**Comment by dash-** [comment]
I love this tool! It's very handy.

Unfortunately, it's also breaking my Cypress tests in my local development environment. I need to be able to programmatically prevent the floating buttons from appearing within Cypress, preferably without removing them completely from my development environment.

**Comment by NiklasBeierl** [comment]
@dash- Does that mean vue-devtools works for you in a cypress environment? As in: You could actually interact with them in a cypress-driven browser? Because that is what I am trying to achieve at the moment. (I guess I am having the opposite of your problem).

**Comment by tryforceful** [comment]
> Currently, DevTools will be forcibly enabled during development (vite dev) and will never be bundled in production (vite build). So, don't worry about that.

(cc @alexzhang1030 ) I think this is worth mentioning in the main documentation, or in the FAQ.  It's not clear (at least to me, and probably other developers) that the plugin doesn't get included in `vite build` steps out-of-the-box without any user configuration.

**Comment by alexzhang1030** [comment]
> > Currently, DevTools will be forcibly enabled during development (vite dev) and will never be bundled in production (vite build). So, don't worry about that.
> 
> 
> 
> (cc @alexzhang1030 ) I think this is worth mentioning in the main documentation, or in the FAQ.  It's not clear (at least to me, and probably other developers) that the plugin doesn't get included in `vite build` steps out-of-the-box without any user configuration.

make sense, I will update it

**Comment by WORMSS** [comment]
Does anyone have a fix for removing this annoying floating panel thing?

![image](https://github.com/user-attachments/assets/4ea336ea-fee8-4a3f-9cf6-3b52922b978f)

I have the option turned off.. but the thing keeps reappearing?

![image](https://github.com/user-attachments/assets/5adf8f78-3deb-4103-870e-259fadb0e727)

Other than doing this little hack.

![image](https://github.com/user-attachments/assets/2ee05121-3a6a-4c39-97d6-2a9ee9643203)

**Comment by lbhopper** [comment]
Comment out `import vueDevTools from 'vite-plugin-vue-devtools'` in vite.config.js then comment out `vueDevTools()` from the plugins
![vite config ts](https://github.com/user-attachments/assets/c685668b-d538-448b-bf85-b993f1b7f74b)

**Comment by SultonbekovSarvarbek** [comment]
the same problem

**Comment by ahsath** [comment]
> ### How to open it again?
> By shortcut
> 
> * `cmd + shift + D` in `macOS`
> * `alt + shift + D` in `Windows`
> 
> align with `Nuxt devtools`.

for me it was 

* `shift + option + D` in `macOS`

**Comment by wuestkamp** [comment]
The option to disable it is no longer available in the settings like @walmon wrote.
Did it move some place else?

**Comment by AimForNaN** [comment]
As of node v25.2.0, it would be really nice to force disable devtools from being used by anything (e.g. vue, pinia, etc), *programmatically*, whether that be in the vite config, env--wherever.

**Comment by da70** [comment]
Jumping onto this older thread which I found when trying to troubleshoot false positives in Playwright HTML tests caused by the HTML and CSS injected into the page by `vite-plugin-vue-devtools`.

Vite's [Using Plugins > Conditional Application](https://vite.dev/guide/using-plugins#conditional-application) linked to in [this older comment ](https://github.com/vuejs/devtools/issues/406#issuecomment-2133875500) does actually work, but not if you strictly follow the structure of the example on that page.  This doesn't work (for `vite-plugin-vue-devtools` version 8.1.1):

```js
import vueDevTools from 'vite-plugin-vue-devtools';
import { defineConfig } from 'vite';

export default defineConfig({
  plugins: [
    {
      ...vueDevTools(),
      apply: 'build',
    },
  ],
})
```

The reason it does not work is because `vueDevTools()` returns an array of 3 elements, two objects and one array:

```js
[
    {
        name: 'vite-plugin-inspect',
        ...[SNIPPED]...
    },
    [
        {
            name: 'vite-plugin-vue-inspector',
            ...[SNIPPED]...
        },
        {
            name: 'vite-plugin-vue-inspector:post',
            ...[SNIPPED]...
        }
    ],
    {
        name: 'vite-plugin-vue-devtools',
        enforce: 'pre',
        apply: 'serve',
        configResolved: [Function: configResolved],
        configureServer: [Function: configureServer],
        resolveId: [AsyncFunction: resolveId],
        load: [AsyncFunction: load],
        transform: [Function: transform],
        transformIndexHtml: [Function: transformIndexHtml],
        buildEnd: [AsyncFunction: buildEnd]
    }
]
```

So the correct way to change `apply` to "build" for `vite-plugin-vue-devtools` is to do this something like this:

```js
const vueDevToolsConfig = vueDevTools();
if ( process.env.DISABLE_VITE_PLUGIN_VUE_DEVTOOLS ) {
    vueDevToolsConfig[ 2 ].apply = 'build';
}

...

export default defineConfig( {
    plugins: [
        vue(),
        vueDevToolsConfig,
    ],
...
};
```

I also added an assertion that `vueDevToolsConfig[ 2 ].name === 'vite-plugin-vue-devtools'` before changing `.apply` and continuing with the tests, because the structure could change in the future when the plugin is upgraded (I didn't see anything in the documentation stating the structure was stable).

Hope this helps.
